### DISCUSSION TITLE
Fit single time series using multivariate approaches

### DISCUSSION THREAD

**Discussion by asgeorges** [winedarksea/AutoTS#122] [Q&A] [answered]
Hi!  

I'm wondering if it's possible to fit just a single time series (or some subset of time series) using multivariate approaches.  

Producing models for every single time series I have has been too time consuming for my purposes - so I'm trying to whittle down the number of time series I'm actually predicting for.  

In other words, basically be able to specify the target variable(s) where all other time series are considered covariates.  

Thanks,
Alex

**Comment by winedarksea** [comment]
A couple of ways you could do this. @asgeorges 

One might be to pass in `weights` to `.fit()` and then pass `subset` to `AutoTS()`. Have extremely large weights for the series you care about (and 0 even for the others). Subsetting is where it trains on only a fraction of series, and it chooses which fraction based on the weights, if given. It might be a bit slow on `.predict` because it will generate predictions on all series, but the model selection part will be much faster.

Another is to use your extra series to create a regressor. You can then forward lag this regressor and use it. I have a function for that:
```python
from autots import create_regressor

# example regressor with some things we can glean from data and datetime index
# note this only accepts `wide` style input dataframes
regr_train, regr_fcst = create_regressor(
    df,
    forecast_length=forecast_length,
    frequency=frequency,
    drop_most_recent=drop_most_recent,
    scale=True,
    summarize="auto",
    backfill="bfill",
    fill_na="spline",
    holiday_countries=["US"],  # requires holidays package
    datepart_method="simple_2",
)
```
note the `summarize` argument, which will attempt to summarize the input series if there are a lot of them.

**Comment by asgeorges** [comment, marked as answer]
So speedy!  

So the first suggestion seems like the reverse of what I want to do: 

- **Suggestion**: use subset of time series for covariates, predict on all series.
- **Wanted**: use all time series for covariates (other than some subset), predict on subset of time series

To be clear about the 2nd suggestion, is this the overall logic:

1. Create a regressor using all variables we consider covariates
2. Construct a df of the single time series we want to predict on
3. Fit an AutoTS model w/ this regressor & this df 

Thanks again.

**Reply by winedarksea** [reply]
@asgeorges I am somewhat skeptical of how well a model can do using 'lots and lots' of covariates. Eventually they might start finding [spurious correlations.](https://www.tylervigen.com/spurious-correlations) And do keep in mind there are plenty of ways to choose faster models (`runtime_weighting`, `model_list='fast'`, etc.) if speed is the main concern.

Second suggestion does sound like it better fits your use case. Yes, that sounds like an accurate summary. Keep in mind there is a `model_mode='regressor'` option that forces use of the regressor, otherwise it will try models both with an without regressors (worth seeing if it finds your regressor helpful or not), and that not all models use regressors, but there's a model_list for that.

**Reply by asgeorges** [reply]
Got it, thanks!
